### PR TITLE
Import the web app to ros2-web-bridge/demo

### PR DESCRIPTION
**Pull request by qiuzhong**

### CODE REVIEW

**Comment by minggangw** [comment]
I am going to review it soon, thanks!

File: demo/js/controller.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    messageType: 'geometry_msgs/Twist'
+  });
+
+  this.log = '';
+}
+
+WebRosController.prototype.sendVelTopic = function(vel) {
+  console.log('send velocity topic');
+
+  if (this.pubVelTopicTimer) {
```
Comment (minggangw) [inline demo/js/controller.js]: Please use the function `shutdownTimer` here to avoid repetition

File: demo/js/log.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  // }, 100);
+
+  setTimeout(() => {
+    document.getElementById('log').innerHTML = logArray.join('<br />');
+  }, 100);
+
+  // setTimeout(() => {
+    // document.getElementById('log').innerHTML = logArray.join('')
+  // }, 200);
+};
```
Comment (minggangw) [inline demo/js/log.js]: Please remove the comments if they will not be uncommented later.

File: demo/js/controller.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      this.currentVelocity.linear.x = -this.currentVelocity.linear.x;
+    }
+    this.sendVelTopic(this.currentVelocity);
+    this.log = 'Publish sensor_msgs/Twist: ' + JSON.stringify(this.currentVelocity);
+  } else {
+    let backVel = {
+      linear: {x: 0.0, y: 0.0, z: 0.0},
+      angular: {x: 0.0, y: 0.0, z: 0.0}
+    };
+    backVel.linear.x = -this.defaultVelocity.linear.x;
```
Comment (minggangw) [inline demo/js/controller.js:165]: I think `Object.create()` is better to be used here, like:
```JavaScript
let backVel = Object.create(zeroVelocity);
backVel.linear.x = -this.defaultVelocity.linear.x;
```

File: demo/js/log.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
+function Logger(logDivId, controller) {
+  this.logDivId = logDivId;
+  this.lineNum = 40;
+  this.title = '';
+  this.rosController = controller;
+
+  this.log = '';
```
Comment (minggangw) [inline demo/js/log.js]: `this.log` may cause confusion, as the variable `log` only stands for the content to be logged here, not something like `console.log` right?

File: demo/js/controller.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  this.pubVelTopicTimer = null;
+  this.pubVelTopicInterval = 100;
+  this.isRobotMoving = false;
+  this.velocityTopic = new ROSLIB.Topic({
+    ros: this.ros,
+    name: '/cmd_vel',
+    messageType: 'geometry_msgs/Twist'
+  });
+
+  this.log = '';
```
Comment (minggangw) [inline demo/js/controller.js]: the same problem as the log.js

**Review by minggangw — COMMENTED** [review]
I found that the text colour of the log output in the web is not very distinct, shall we change a colour which can be distinguished with the background easily.

File: demo/js/controller.js (hunk 1 above)
Comment (qiuzhong) [reply demo/js/controller.js]: Sure. I'll update this.

File: demo/js/log.js (hunk 2 above)
Comment (qiuzhong) [reply demo/js/log.js]: This should be removed. Besides, there is other commented content to be removed.

File: demo/js/controller.js (hunk 3 above)
Comment (qiuzhong) [reply demo/js/controller.js:165]: Actually, `Object.create()` or `Object.assign` is shallow copy. See this:
```
> let zero = {linear: {x: 0.0, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}};
undefined
> let backVel = Object.create(zero);
undefined
> backVel.linear.x = -0.05;
-0.05
> zero
{ linear: { x: -0.05, y: 0, z: 0 },
  angular: { x: 0, y: 0, z: 0 } }
```

File: demo/js/log.js (hunk 4 above)
Comment (qiuzhong) [reply demo/js/log.js]: Right.
The log.js is totally a mess. I think it needs to be rewritten.

File: demo/js/controller.js (hunk 5 above)
Comment (qiuzhong) [reply demo/js/controller.js]: I'll rewrite the log.js

File: demo/js/controller.js (hunk 3 above)
Comment (minggangw) [reply demo/js/controller.js:165]: Oops, my error. It should be `backVel.linear = {x: -0.05, y: 0.0, z: 0.0}`, my thought is that using 'Object.create` to simplify the definition of variable.

**Comment by qiuzhong** [comment]
@minggangw , I updated the demo app.

**Comment by minggangw** [comment]
Got it, I am going to have a look at it soon. Thanks!

**Review by minggangw — APPROVED** [review]
LGTM, thanks for writing this concise demonstration for the web bridge, which is helpful and useful to the beginners.

**Pull request merged**